lib/smtp-server.js

```diff
@@ -170,6 +170,9 @@ class SMTPServer extends EventEmitter {
                     connection.close();
                 });
             }
+            if (typeof callback === 'function') {
+              return callback();
```

Comment:
I can see why you added this but it can result in callback fired twice (once here and once the server is actually closed). You should probably clear the `callback` value so if the same block is ran after server close event then `typeof callback === 'function'` would not be true anymore.

Probably something like this:
```js
let realCallback = callback;
callback = null;
return realCallback();
```

Reply:
I agree with your comment.  I've changed the code according to your suggestion.